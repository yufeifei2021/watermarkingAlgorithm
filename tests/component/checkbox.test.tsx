import { cleanup, fireEvent, render } from '@testing-library/react';
import CheckboxWithLabel from './Checkbox';

afterEach(cleanup);

it('CheckboxWithLabel changes the text after click', () => {
  const { queryByLabelText, getByLabelText, baseElement } = render(
    <CheckboxWithLabel labelOn="On" labelOff="Off" />,
  );
  expect(baseElement).toMatchSnapshot();
  expect(getByLabelText(/Off/i)).toBeInTheDocument();
  fireEvent.click(getByLabelText(/Off/i));
  expect(queryByLabelText(/On/i)).toBeTruthy();
});
