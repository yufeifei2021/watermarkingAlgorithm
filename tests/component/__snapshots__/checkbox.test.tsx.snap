// Vitest Snapshot v1

exports[`CheckboxWithLabel changes the text after click 1`] = `
<body>
  <div>
    <div>
      <input
        id="test"
        type="checkbox"
      />
      <label
        for="test"
      >
        Off
      </label>
    </div>
  </div>
</body>
`;
